### PR TITLE
:sparkles: (go/v4) Add Metrics Validation and Helper Functions to E2E Tests

### PR DESCRIPTION
**Pull request by camilamacedo86**
Provide further improvements for e2e tests test to help users be aware of how to tests using the metrics endpoint and validate if the metrics are properly expose. Furthermore, shows how to use it to validate the reconciliation. 

Partially closes: https://github.com/kubernetes-sigs/kubebuilder/issues/4120

### CODE REVIEW

**Comment by camilamacedo86** [comment]
@Adembc 

With this one I think we can address all.

**Comment by camilamacedo86** [comment]
/hold false

It seems great to fly 🚀

File: testdata/project-v4-multigroup/test/e2e/e2e_test.go [hunk 1]
```diff
@@ -110,5 +126,163 @@ var _ = Describe("controller", Ordered, func() {
...
+func serviceAccountToken() (string, error) {
+	const tokenRequestRawString = `{
+		"apiVersion": "authentication.k8s.io/v1",
+		"kind": "TokenRequest"
+	}`
+
+	// Temporary file to store the token request
+	secretName := fmt.Sprintf("%s-token-request", serviceAccountName)
+	tokenRequestFile := filepath.Join("/tmp", secretName)
+	err := os.WriteFile(tokenRequestFile, []byte(tokenRequestRawString), os.FileMode(0o755))
```
Comment (mogsie) [inline testdata/project-v4-multigroup/test/e2e/e2e_test.go:235]: 0o644 should be enough, this file does not need the executable bit?

File: pkg/plugins/golang/v4/scaffolds/internal/templates/test/e2e/suite.go [hunk 2]
```diff
@@ -83,6 +83,9 @@ func TestE2E(t *testing.T) {
 }
 
 var _ = BeforeSuite(func() {
+	By("Ensure that Prometheus is enable")
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply pkg/plugins/golang/v4/scaffolds/internal/templates/test/e2e/suite.go:86]: done

File: testdata/project-v4-multigroup/test/e2e/e2e_test.go (hunk 1 above)
Comment (camilamacedo86) [reply testdata/project-v4-multigroup/test/e2e/e2e_test.go:235]: done

**Comment by mogsie** [comment]
/lgtm

(not that I think I have a say in the matter :smile:)

**Pull request closed without merging**